proto/soci.proto

```diff
@@ -28,7 +29,14 @@ message Artifact {
 }
 
 message GetArtifactsRequest {
-  string image_name = 1;
+  string image = 1;
+
+  // If an image has multiple variants, this selector is used
+  // to select a single variant.
+  registry.Platform platform = 2;
```

Comment:
feel free to copy over these two messages into this file. that way we can delete `registry.proto` when you're done.

Reply:
Because the code using these is going to be interacting with the stuff in registry.go which uses the registry.proto versions, it's actually easier to have a shared definition, so I'll just leave them in there for now and we can move them in the future if/when it makes sense.